public/app/features/alerting/unified/mocks/plugins.ts

```diff
@@ -1,12 +1,28 @@
 import { http, HttpResponse } from 'msw';
 
 import { PluginMeta } from '@grafana/data';
+import { config } from '@grafana/runtime';
 import { plugins } from 'app/features/alerting/unified/testSetup/plugins';
 
-export const pluginsHandler = (pluginsArray: PluginMeta[] = plugins) =>
-  http.get<{ pluginId: string }>(`/api/plugins/:pluginId/settings`, ({ params: { pluginId } }) => {
+/**
+ * Returns a handler that maps from plugin ID to PluginMeta, and additionally sets up necessary
+ * config side effects that are expected to come along with this API behaviour
+ */
+export const pluginsHandler = (pluginsArray: PluginMeta[] = plugins) => {
+  plugins.forEach(({ id, baseUrl, info, angular }) => {
```

Comment:
Interested to hear thoughts on this. It ties the plugin handler setup along with the configuration of the `config` module, but the two kind of _should_ go hand in hand?

Reply:
I think it's fine – I've done similar things where the handler also spies on / mocks  the runtime config :)